devicetypes/smartthings/ecobee-thermostat.src/ecobee-thermostat.groovy

```diff
@@ -156,47 +156,49 @@ void poll() {
 def generateEvent(Map results) {
 	log.debug "parsing data $results"
 	if(results) {
-		results.each { name, value ->
 
 			def linkText = getLinkText(device)
-			def isChange = false
-			def isDisplayed = true
+		def supportedThermostatModes = []
+		def thermostatMode = null
+
+		results.each { name, value ->
 			def event = [name: name, linkText: linkText, descriptionText: getThermostatDescriptionText(name, value, linkText),
 						 handlerName: name]
 
 			if (name=="temperature" || name=="heatingSetpoint" || name=="coolingSetpoint" ) {
 				def sendValue =  location.temperatureScale == "C"? roundC(convertFtoC(value.toDouble())) : value.toInteger()
-				isChange = isTemperatureStateChange(device, name, value.toString())
-				isDisplayed = isChange
-				event << [value: sendValue, unit: temperatureScale, isStateChange: isChange, displayed: isDisplayed]
+				event << [value: sendValue, unit: temperatureScale]
 			}  else if (name=="maxCoolingSetpoint" || name=="minCoolingSetpoint" || name=="maxHeatingSetpoint" || name=="minHeatingSetpoint") {
 				def sendValue =  location.temperatureScale == "C"? roundC(convertFtoC(value.toDouble())) : value.toInteger()
 				event << [value: sendValue, unit: temperatureScale, displayed: false]
 			}  else if (name=="heatMode" || name=="coolMode" || name=="autoMode" || name=="auxHeatMode"){
-				isChange = isStateChange(device, name, value.toString())
-				event << [value: value.toString(), isStateChange: isChange, displayed: false]
+				if (value == true) {
+					supportedThermostatModes << ((name == "auxHeatMode") ? "auxheatonly" : name - "Mode")
+				}
+				return // as we don't want to send this event here, proceed to next name/value pair
 			}  else if (name=="thermostatFanMode"){
-				isChange = isStateChange(device, name, value.toString())
-				event << [value: value.toString(), isStateChange: isChange, displayed: false]
+				sendEvent(name: "supportedThermostatFanModes", value: fanModes(), displayed: false)
```

Comment:
I'm actually not, it needs to be sent once but if I add it to installed or a place where it is only called when the DTH is installed we miss all existing DTHs so I made this intentionally this time. It could potentially be moved though after this implementation have been soaking for some time.